include/log0recv.h

```diff
@@ -205,8 +205,18 @@ struct recv_addr_struct {
   hash_node_t addr_hash;
 };
 
+struct Space_id_page_no_hash {
+  std::size_t operator()(const std::pair<space_id_t, page_no_t> &pair) const {
```

Comment:
Make this into a new type, it's used everywhere. We can make the other changes later.

struct Page_id {
  space_id_t m_space;
  page_no_t m_page_no;
};

Overload the <, ==, > and != operator too and add is_null() and to_string() method to it.